### ISSUE TITLE
Standardise the 'default' alias of materials

### ISSUE THREAD

**Issue by Moderocky** [SkriptLang/Skript#2091]
### Description

Currently, when you use `"%target block%"` or `"%tool%"` or something similar (returning the string-name of a material) these are very inconsistent.

For example, coral blocks are all `living <colour> coral block` apart from the blue one, which is `living tube coral blocks`.
However, the fronds/fans use _different_ alias patterns. Most of these use the actual type name by default (tube/brain) whatever, but not all.

These inconsistencies are also present in other things, such as music discs and some of the less-frequently-used items.

However, correcting these would only be a small part of the problem.

An item's Skript alias doesn't always contain the default minecraft ID as a possible pattern. This makes things complicated as you have to check them manually to find whatever weird combination is used.

What I am proposing is that the default alias (what's returned in "%tool%") return the material's Minecraft ID, (and if it's a blockstate/variation then that should be appended).

For example:
- Stone = `stone`
- Diamond hoe = `diamond_hoe`
- Tube coral = `tube_coral_block`
etc.

Where this pattern isn't already an alias, it should definitely be added.

This would simplify stuff, make Skript more intuitive and easier, as well as clearing up all of the little issues and strange inconsistencies that cause problems for some people.

**Comment by bensku** [comment]
Minecraft ids sometimes change between versions and are not always logical. We should avoid presenting them to scripts whenever possible. Instead, we should ensure that all materials have default aliases. @TheBentoBox might have some idea how to do this.

Material names returned by Skript are guaranteed (except when there are bugs) to be the closest possible aliases. When there are multiple equally good options, Skript randomly picks one of them. This is not ideal, and I'd like to change it in future.

**Comment by TheBentoBox** [comment]
Something as simple as always picking the first viable alias instead of a random one would help. Ideally we'd be able to more finely control this but it's tricky because we're not actually manually writing out every single alias, but rather using variations to build the entire tree of possible states for that material. What I mean is: if I was manually writing `waterlogged open top-half westward oak trapdoor`, I could mark that as `[isDefault=true]` in a tag or something that the aliases could recognize. But again, that isn't the case.

Realistically the issue is that the default string representation is almost *too* specific. It's great to be able to specify all of those states for placing blocks or doing specific comparisons, but generally someone looking at a waterlogged open top-half westward oak trapdoor only really needs to see "oak trapdoor". But how to best do that is also unclear; you can say that the aliases should then have support for returning a string with none of the variations applied, but sometimes those variations are a core part of the alias definition. For example, the wood type of the trapdoor is a variation which both prepends the wood type and modifies the Minecraft ID itself. Stripping that would just return "trapdoor" which isn't really that helpful or accurate.

tl;dr I have vague ideas but there's a reason this hasn't been done yet. There really isn't a great solution. And I see it as encouragement to not work directly with the strings anyway.

**Comment by Moderocky** [comment]
True, though I wasn't thinking of it in terms of strings, it's more that I'd like to be able to think "this item is called `stone_hoe` in Minecraft, and so I know that I can use `stone_hoe` in Skript as well", for example.

Obviously, `stone hoe` is the same in Skript, but some others aren't the same and I think that confuses some people.
Especially since there isn't a generated list of aliases anymore that I can easily search through.

**Comment by TheBentoBox** [comment]
You're describing the same issue though. The problem isn't that we're randomly defining aliases differently, it's that Skript doesn't consistently return the aliases formatted in the same when there's a bunch of grouped items referencing an alias variation.
